src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanOrchestrator.hpp

```diff
@@ -223,7 +223,19 @@ class TScanOrchestrator final : public TOSPrimitives
         }
         else if (message->type() == BufferType::BufferType_JSON)
         {
-            auto jsonData = nlohmann::json::parse(message->data()->data());
+            nlohmann::json jsonData;
+            try
+            {
+                jsonData =
+                    nlohmann::json::parse(message->data()->data(), message->data()->data() + message->data()->size());
+            }
+            catch (const nlohmann::json::parse_error& e)
+            {
+                std::string jsonMessage {message->data()->data(), message->data()->data() + message->data()->size()};
+                logDebug2(WM_VULNSCAN_LOGTAG, "Error parsing JSON: %s", jsonMessage.c_str());
+                throw;
+            }
+
```

Comment:
```suggestion
            nlohmann::json jsonData = nlohmann::json::parse(message->data()->data(), message->data()->data() + message->data()->size());
```

I think the try-catch is not necessary and this propagates to the error handling already available.

Both in the initEventDelayedDispatcher lambda, as well as in the initEventDispatcher, In both places the exception is caught, the event dropped/not processed and a log is printed.

Reply:
Hi.

The idea here was to obtain more information, because the current log doesn't show the full JSON message

```
ERROR: Error processing delayed event: [json.exception.parse_error.101] parse error at line 1, column 73: syntax error while parsing value - invalid literal; last read: '"no-index":true}<U+0004>'; expected end of input
```

So I catch and re-throw but log in the middle.
Another option is to use 


```c++

nlohmann::json jsonData = nlohmann::json::parse(message->data()->data(), message->data()->data() + message->data()->size(), nullptr, false);
```

Then we check if it has been discarded

```c++
if(jsonData.is_discarded()) {
```

And finally we log and throw.
This approach generates a more user friendly message and it keeps showing more context to the developer. 
The current one isn't enough in case of errors for the user to understand what is happening and for us to debug the problem.

Reply:
If you want to have an additional log, let's do it where appropriate.
what was said, extend in initEventDelayedDispatcher and in initEventDispatcher.

Let's not use the function that does not throw exceptions, nor let's catch exceptions elsewhere.

Reply:
Now I understand.
I'll move it to the right place and I'll add a catch for `nlohmann::json::exception` in the `initEventDelayedDispatcher`.
Thank you.

Reply:
Now I understand.
I'll move it to the right place and I'll add a catch for `nlohmann::json::exception` in the `initEventDelayedDispatcher`.
Thank you.

Reply:
Changes uploaded.